Modules/GUI/UnifiedGUI.lua

```diff
@@ -0,0 +1,221 @@
+-- ------------------------------- --
+local  _, CLM = ...
+-- ------ CLM common cache ------- --
+local LOG       = CLM.LOG
+local CONSTANTS = CLM.CONSTANTS
+local UTILS     = CLM.UTILS
+-- ------------------------------- --
+
+local pairs = pairs
+local tinsert = table.insert
+
+-- Libs
+
+local AceGUI = LibStub("AceGUI-3.0")
+local AceConfigRegistry = LibStub("AceConfigRegistry-3.0")
+local AceConfigDialog = LibStub("AceConfigDialog-3.0")
+
+local REGISTRY = "clm_unifiedgui_gui_options"
+
+local function InitializeDB(self)
+    self.db = CLM.MODULES.Database:GUI('unifiedgui', {
+        location = {nil, nil, "CENTER", 0, 0 }
+    })
+end
+
+local function StoreLocation(self)
+    self.db.location = { self.aceObjects.top:GetPoint() }
+end
+
+local function RestoreLocation(self)
+    if self.db.location then
+        self.aceObjects.top:ClearAllPoints()
+        self.aceObjects.top:SetPoint(self.db.location[3], self.db.location[4], self.db.location[5])
+    end
+end
+
+local UnifiedGUI = { tabs = {} }
+function UnifiedGUI:Initialize()
```

Comment:
I'm not a fan of this pattern. Essentially you're creating a singleton table called `UnifiedGUI`, but you've made it someone else's responsibility to initialize it.
Either you initialize it on file load, or you initialize it on some event (and you register the listener on file load), or the cleanest: you initialize it automatically whenever calling any function that requires it to be initialized.

The problem with this pattern is that your code is now littered in every public function with something like `if not self._initialized then return end`. Which makes those public APIs weird to use (since I have to check the return value type) and it gives me little recourse; so then any callee now needs to understand that maybe calling `UnifiedGUI:Initalize()` will solve their problems. But that function has prerequisites which they are not aware of etc...

From what I understand this code has a few requirements:
- LedgerManager
- Database

Imo just check if these are set on file load.

Reply:
Separated into new issue